tests/integration/awslambda/test_lambda_api.py

```diff
@@ -573,6 +574,110 @@ def test_list_functions(self, create_lambda_function, lambda_su_role, snapshot,
         snapshot.match("list_all", list_all)
         snapshot.match("list_default", list_default)
 
+    def test_vpc_config(
+        self, create_lambda_function, lambda_su_role, snapshot, aws_client, cleanups
+    ):
+        """
+        Test "VpcConfig" Property on the Lambda Function
+
+        Note: on AWS this takes quite a while since creating a function with VPC usually takes at least 4 minutes
+
+        TODO: test a few more edge cases (e.g. multiple subnets / security groups, invalid vpc ids, etc.)
+        """
+
+        # VPC setup
+        security_group_name_1 = f"test-security-group-{short_uid()}"
+        security_group_name_2 = f"test-security-group-{short_uid()}"
+        vpc_id = aws_client.ec2.create_vpc(CidrBlock="10.0.0.0/16")["Vpc"]["VpcId"]
+        cleanups.append(lambda: aws_client.ec2.delete_vpc(VpcId=vpc_id))
+        aws_client.ec2.get_waiter("vpc_available").wait(VpcIds=[vpc_id])
+        security_group_id_1 = aws_client.ec2.create_security_group(
+            VpcId=vpc_id, GroupName=security_group_name_1, Description="Test security group 1"
+        )["GroupId"]
+        cleanups.append(lambda: aws_client.ec2.delete_security_group(GroupId=security_group_id_1))
+        security_group_id_2 = aws_client.ec2.create_security_group(
+            VpcId=vpc_id, GroupName=security_group_name_2, Description="Test security group 2"
+        )["GroupId"]
+        cleanups.append(lambda: aws_client.ec2.delete_security_group(GroupId=security_group_id_2))
+        subnet_id_1 = aws_client.ec2.create_subnet(VpcId=vpc_id, CidrBlock="10.0.0.0/24")["Subnet"][
+            "SubnetId"
+        ]
+        cleanups.append(lambda: aws_client.ec2.delete_subnet(SubnetId=subnet_id_1))
+        subnet_id_2 = aws_client.ec2.create_subnet(VpcId=vpc_id, CidrBlock="10.0.1.0/24")["Subnet"][
+            "SubnetId"
+        ]
+        cleanups.append(lambda: aws_client.ec2.delete_subnet(SubnetId=subnet_id_2))
+        snapshot.add_transformer(snapshot.transform.regex(vpc_id, "<vpc_id>"))
+        snapshot.add_transformer(snapshot.transform.regex(subnet_id_1, "<subnet_id_1>"))
+        snapshot.add_transformer(snapshot.transform.regex(subnet_id_2, "<subnet_id_2>"))
+        snapshot.add_transformer(
+            snapshot.transform.regex(security_group_id_1, "<security_group_id_1>")
+        )
+        snapshot.add_transformer(
+            snapshot.transform.regex(security_group_id_2, "<security_group_id_2>")
+        )
+
+        cleanups.append(
+            lambda: aws_client.awslambda.delete_function(FunctionName=function_name)
+        )  # needed because otherwise VPC is still linked to function and deletion is blocked
```

Comment:
Unfortunately this still doesn't work properly on AWS and the last subnet/security group + vpc are leaking. Not critical for this PR but we probably need to wait for the lambda to actually  disappear before this is detached an can be deleted :thinking:

Reply:
VPC testing is annoying 🐌 

nit: would be good to add the comment in the test such that it does not get lost in this PR discussion
💡 slow tests + improper cleanup are candidates for test labels